### DISCUSSION TITLE
AppRail DX improvement

### DISCUSSION THREAD

**Discussion by unlocomqx** [skeletonlabs/skeleton#1396] [General]
Hi,

I'd like to suggest a few improvements to the `AppRail` and `AppRailTile` components

I'll start with the "final" code

```svelte
<script>
  import { AppRail, AppRailTile } from "@skeletonlabs/skeleton"
  import { derived } from "svelte/store"
  import { page } from "$app/stores"

  const selected = derived(page, $page => $page.url.pathname)
</script>

<AppRail {selected}>
  <AppRailTile tag="a" href="/about">Tile 1</AppRailTile>
  <AppRailTile tag="a" href="/account">Tile 2</AppRailTile>
  <AppRailTile tag="a" href="/login">Tile 3</AppRailTile>
</AppRail>
```
Here are my suggestions to make this code possible (Currently you need more verbose code to handle state)
- Take the `value` from `href` when no `value` is passed (or when tag=a)
- Don't [call](https://github.com/skeletonlabs/skeleton/blob/dev/src/lib/components/AppRail/AppRailTile.svelte#L40) .set on the store when it's a readable like in this case

This code is much simpler then the code shown in the [docs](https://www.skeleton.dev/components/app-rail#page-state)

If you like my suggestion, I can submit a PR and then we'll see how it goes

Cheers

**Comment by Sarenor** [comment]
Would at least need another prop to be able to specify the active class.
I think the current way is more adaptive and better fits the idea of providing building blocks for arbitrary logic.

On the other hand, nothing in your proposed changes would break the current way of doing things and just simplify the - probably - most common usecase, so I'm definitely not against it.

**Comment by unlocomqx** [comment]
The active class can be [passed](https://github.com/skeletonlabs/skeleton/blob/dev/src/lib/components/AppRail/AppRail.svelte#L24) normally like we used to

My suggestions won't change your current user code, it just simplifies the most common use case (I think)

**Reply by Sarenor** [reply]
Should have looked at the code more closely, yeah that'd work, I think.
Let's wait for @endigo9740 to weigh in, but in general I see nothing that speaks against it

**Comment by endigo9740** [comment]
@unlocomqx

> Take the value from href when no value is passed (or when tag=a)

I agree, the explicit tag is redundant when an `href` is available to determine this. 

> Don't [call](https://github.com/skeletonlabs/skeleton/blob/dev/src/lib/components/AppRail/AppRailTile.svelte#L40) .set on the store when it's a readable like in this case

I'm not immediately clear on the benefit of this, can you expand on this?

**Reply by unlocomqx** [reply]
> I'm not immediately clear on the benefit of this, can you expand on this?

Not a benefit per se, it's just to avoid the error when a readable is passed.

In this case a readable is used to pass the current page url.

**Reply by endigo9740** [reply]
Ok, well submit a PR and we'll review. Thanks!

**Reply by unlocomqx** [reply]
Thanks, I will create a PR soon

Cheers
